packages/core/src/App/Containers/PasswordSelectionModal/password-selection-modal.jsx

```diff
@@ -20,85 +20,91 @@ const PasswordSelectionModal = ({
-                <div className='account-signup__error-wrapper'>
-                    <Button
-                        large
-                        secondary
-                        text={localize('Cancel')}
-                        type='button'
-                        onClick={() => isModalVisible(false)}
-                    />
-                    <Button
-                        large
-                        primary
-                        text={localize('Create new account')}
-                        type='button'
-                        onClick={() => redirectToSignUp({ is_appstore })}
-                    />
-                </div>
-            </React.Fragment>
-        ) : (
-            <Button
-                className={classNames('account-signup__btn', {
-                    'account-signup__btn--disabled': !values.password || errors.password || isSubmitting,
-                })}
-                id='dt_core_account-signup-modal_submit-btn'
-                type='submit'
-                is_disabled={!values.password || !!errors.password || isSubmitting}
-                text={localize('Start trading')}
-                large
-                primary
-            />
-        )}
-    </div>
-);
+            {api_error ? (
+                <React.Fragment>
+                    <Text
+                        as='p'
+                        size='xxs'
+                        color='loss-danger'
+                        className='account-signup__subtext account-signup__subtext--error'
+                        align='center'
+                    >
+                        {api_error}
+                    </Text>
+                    <div className='account-signup__error-wrapper'>
+                        <Button
+                            large
+                            secondary
+                            text={localize('Cancel')}
+                            type='button'
+                            onClick={() => isModalVisible(false)}
+                        />
+                        <Button
+                            large
+                            primary
+                            text={localize('Create new account')}
+                            type='button'
+                            onClick={() => redirectToSignUp({ is_appstore })}
+                        />
+                    </div>
+                </React.Fragment>
+            ) : (
+                <Button
+                    className={classNames('account-signup__btn', {
+                        'account-signup__btn--disabled': !values.password || errors.password || isSubmitting,
+                    })}
+                    id='dt_core_account-signup-modal_submit-btn'
+                    type='submit'
+                    is_disabled={!values.password || !!errors.password || isSubmitting}
+                    text={localize('Start trading')}
+                    large
+                    primary
+                    onClick={() => {
```

Comment:
this is the only change in this file, right? if yes revert the changes and add this line to the code

Reply:
If i add the setisPreAppstore flag here, it doesnt work properly. I would have to add it in onboarding. then its fine

Reply:
No I mean you do not need to add `{ return (` in line 24

Reply:
Oh... alrights